tcf_website/views/browse.py

```diff
@@ -213,67 +214,66 @@ def course_instructor(request, course_id, instructor_id):
-    avg_gpa = safe_round(instructor.average_gpa())
-
-    courses = group_by_dept(instructor, instructor.get_courses())
-    return render(
-        request, 'instructor/instructor.html', {
-            'instructor': instructor,
-            'avg_rating': avg_rating,
-            'avg_difficulty': avg_difficulty,
-            'avg_gpa': avg_gpa,
-            'courses': courses
-        })
-
-
-def group_by_dept(instructor, courses):
-    """ Group instructor's courses by subdepartment.
-
-        Returns a dictionary mapping subdepartment names to ids and
-        lists of Course data.
-    """
-    subdept_ids = list(
-        set(courses.values_list('subdepartment', flat=True)))
-
-    # Contains IDs and names of all subdepartments for Instructor's Courses
-    subdepts = sorted(
-        list(
-            map(
-                lambda subdept_id: [
-                    subdept_id,
-                    Subdepartment.objects.get(
-                        pk=subdept_id).name],
-                subdept_ids)),
-        key=lambda x: x[1])
-
-    # Create dictionary corresponding dept to courses
-    grouped_courses = {}
-    for subdept in subdepts:
-        grouped_courses[subdept[1]] = {
-            "id": subdept[0],
-            "courses": []
-        }
-
-    for course in courses:
-        course_subdept = course.subdepartment.name
-        course_data = {
-            # autopep8 makes the formatting for this wack, sorry
-            'name': str(course), 'id': course.id,
-            'avg_rating': safe_round(
-                Instructor.average_rating_for_course(
-                    instructor, course)),
-            'avg_difficulty': safe_round(
-                Instructor.average_difficulty_for_course(
-                    instructor, course)),
-            'avg_gpa': safe_round(Instructor.average_gpa_for_course(instructor, course)),
-            'last_taught': str(course.semester_last_taught)}
-        grouped_courses[course_subdept]['courses'].append(course_data)
-
-    return grouped_courses
+    instructor: Instructor = get_object_or_404(Instructor, pk=instructor_id)
+    stats: Dict[str, float] = Instructor.objects\
+        .filter(pk=instructor_id)\
+        .prefetch_related('review_set')\
+        .aggregate(
+            avg_gpa=Avg('courseinstructorgrade__average'),
+            avg_difficulty=Avg('review__difficulty'),
+            avg_rating=(
+                Avg('review__instructor_rating') +
+                Avg('review__enjoyability') +
+                Avg('review__recommendability')
+            ) / 3)
+    course_fields: List[str] = ['name', 'id', 'avg_rating', 'avg_difficulty',
+                                'avg_gpa', 'last_taught']
+    queryset: List[Dict[str, Any]] = Course.objects\
+        .filter(section__instructors=instructor, number__gte=1000)\
+        .prefetch_related('review_set')\
+        .annotate(
+            dept=F('subdepartment__name'),
```

Comment:
Do you think it's important to make the distinction between Department and Subdepartment clear? This has been an issue for a while and it might be outside the scope of this PR, but as you know we mix them a bunch (I've been guilty of this a lot lol) and that might not be fantastic.

Reply:
I tried not to change anything in the template, but I agree that subdepartment is a better choice. The full name was not available (`ValueError: The annotation 'subdepartment' conflicts with a field on the model.`), so I used `subdept`, which is already used in `tcf_website/static/reviews/new_review.js` and `tcf_website/views/search.py`. Also fixed in 001b6592654e68ce1811a4652c6a21c4e4866f46

Reply:
Maybe try `subdepartment_name` or `subdept_name`? Kinda unimportant cz it's just used within this view, though.

Reply:
Renamed to `subdepartment_name` in fc412e6f3dcaedb15a25e27a359bb88af2075187.